src/components/List.js

```diff
@@ -132,11 +82,113 @@ const List = props => {
           stats={stats}
           patients={patients}
           municipalities={municipalities}
-          perHospitalChanges={perHospitalChanges}
+          perHospitalChanges={perHospitalChangesWithLongName}
         />
         <Outro />
       </section>
     </div>
   );
 };
 export default List;
+
+const isUndefined = val => val === undefined;
+
+// prepare hospitalsDict
+function prepareHospitalsDict(hospitalsList) {
+  return hospitalsList.map(hospital => [hospital.code, hospital.name]);
+}
+
+// prepare perHospitalChanges
+function getPerHospitalChanges(patients) {
+  const patientsData = patients[patients.length - 1];
+  const patientsDataIsNotUndefined = !isUndefined(patientsData);
+
+  return (
+    patientsDataIsNotUndefined &&
+    Object.entries(patients[patients.length - 1].facilities)
+  );
+}
+
+function findAndPushLongHospitalName(perHospitalChanges, hospitalsDict) {
+  return perHospitalChanges.map(hospital => {
+    const hospitalLongName = hospitalsDict.filter(
+      item => hospital[0] === item[0]
+    )[0][1];
+    return [...hospital, hospitalLongName];
+  });
+}
+
+// get date for Intro component
+function formatToLocaleDateString(
+  formatStr = 'd.M.yyyy',
+  options = { locale: sl }
+) {
+  return dateAsText => {
+    const date = new Date(dateAsText);
+    return format(date, formatStr, options);
+  };
+}
+
+/**
+ * get date with time 0:00:00; need to calculate if fetched data is not up to date
+ * date received is part of an object with properties: year, month, day
+ * at the moment we need to compare 4 dates from fetched data to today
+ */
+function getDateNoTime(obj) {
+  // TODO error check?
+  if (!obj) {
+    const today = new Date();
+    const todayArray = [today.getFullYear(), today.getMonth(), today.getDate()];
+    return new Date(...todayArray);
+  }
+
+  let { year, month, day } = obj;
+  return new Date(year, month - 1, day);
+}
+
+/**
+ * paint red if data is not updated for the current day;
+ * params <somethin>Check are used as className
+ * TODO figure out if you can use date-fns
+ */
+function getChecks({ stats, municipalities, patients, summary }) {
+  // data - no need to destructure summary while it's an object
+  const patientsData = patients[patients.length - 1];
+  const statsData = stats[stats.length - 1];
+  const municipalitiesData = municipalities[municipalities.length - 1];
+
+  // dates
+  const todayDate = getDateNoTime();
+  const patientsDate = getDateNoTime(patientsData);
+  const statsDate = getDateNoTime(statsData);
+  const municipalitiesDate = getDateNoTime(municipalitiesData);
+  const summaryDate = getDateNoTime(summary.testsToday);
```

Comment:
I was just following the Matej's logic.

This is initial code:

```javascript
let check_first =''

const todayDate = parseInt(
    new Date().getFullYear().toString() +
      (new Date().getMonth() + 1).toString() +
      new Date().getDate().toString()
  );

  const patientsDate =
    patients[patients.length - 1].year.toString() +
    patients[patients.length - 1].month.toString() +
    patients[patients.length - 1].day.toString();
// ... same pattern for other dates

if (todayDate - summaryDate === -1) {
    check_first = 'red';
}

if (todayDate - patientsDate > 0) {
    check_second = 'red';
}

if (
    stats[stats.length - 2].statePerAgeToDate[0].allToDate === undefined ||
    todayDate - statsDate > 0
) {
    check_third_age = 'red';
}

if (todayDate - municipalitiesDate > 1) {
  check_third_mun = 'red';
}
```
and because
```javascript
new Date() - new Date(2021,0,17) > 0 === true
```
it was the easiest way for me to follow the logic and not break it.

Reply:
@matejmeglic 

Is there a case when `summaryDate` is greater than `todayDate`?
```javascript
if (todayDate - summaryDate === -1) {
    check_first = 'red';
}
```